### PR TITLE
Consistent with node-semver

### PR DESCRIPTION
**Pull request by milk1000cc**
This PR adds test cases and edge case handling to the Build::Utils#fix_version_string method when version string is like `>1.4.x` or `<=1.4.x` in source Bower packages.

Please see also #263.

### CODE REVIEW

File: app/models/build/utils.rb [hunk 1]
```diff
@@ -52,7 +52,18 @@ def fix_version_string(version)
...
+          case specifier
+          when '>'
+            new[-1] = (new[-1].to_i + 1).to_s
+            new.gsub specifier, '>='
+          when '<='
+            new[-1] = (new[-1].to_i + 1).to_s
+            new.gsub specifier, '<'
+          else
+            new
+          end
```
Comment (joshjordan) [inline app/models/build/utils.rb:66]: How does this behave for cases where the previous version segment is multiple digits (e.g. `<= 2.99.x`)?

File: app/models/build/utils.rb [hunk 2]
```diff
@@ -52,7 +52,18 @@ def fix_version_string(version)
         semVerReg = /#{Regexp.escape(specifier)}\s*(\d+\.)*[x\*]/
 
         version.gsub!(semVerReg) do |match|
-          match.strip[0..-3]
+          new = match.strip[0..-3]
```
Comment (joshjordan) [inline app/models/build/utils.rb:55]: May we use an identifier other than `new`? To my eyes, this creates confusion about whether we're talking about a constructor or an object when I'm scanning quickly.

**Comment by joshjordan** [comment]
Great work, and thanks for the thorough test cases! I had a few comments. Going to look into the Travis CI failures now.

**Comment by joshjordan** [comment]
I really doubt this change is what's breaking that integration test. I won't consider that blocking for this PR.
